File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<vector<int>> levelOrder(TreeNode* root) {
        vector<vector<int>> ans;
        if(root==NULL) return {};
        queue<TreeNode*>q;
        q.push(root);
        q.push(NULL);
        vector<int> v;
        while(!q.empty()) {
            TreeNode* t = q.front();
            
            q.pop();
            if(t==NULL) {
                ans.push_back(v);
                v = {};
                if(!q.empty()) q.push(NULL);
            } else {
                v.push_back(t->val);
                if(t->left)
                    q.push(t->left);
                if(t->right)
                    q.push(t->right);
                
            }
 
        }
        return ans;
    }
};

File: 102-binary-tree-level-order-traversal/binary-tree-level-order-traversal.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def levelOrder(self, root: Optional[TreeNode]) -> List[List[int]]:
        l = []
        ans = []
        if root==None:
            return []
        l.append(root)
        while len(l)!=0:
            p = len(l)
            z = []
            while p:
                p-=1
                k = l[0]
                z.append(k.val)
                l.pop(0)
                if k.left:
                    l.append(k.left)
                if k.right:
                    l.append(k.right)
            ans.append(z)
        return ans

        